activerecord/test/cases/base_test.rb

```diff
@@ -1518,4 +1518,15 @@ def test_default_values_are_deeply_dupped
     assert Developer.new.respond_to?(:last_name=)
     assert Developer.new.respond_to?(:last_name?)
   end
+
+  test "when #reload called, ignored columns' attribute methods are not defined" do
+    developer = Developer.create!(name: 'Developer')
+    refute developer.respond_to?(:first_name)
+    refute developer.respond_to?(:first_name=)
+
+    developer.reload
```

Comment:
If we introduce changes to `build_select`, should we also assert that `to_sql` doesn't include the ignored column?